### PR TITLE
ShortcutDialog improvements.

### LINKED ISSUE #784: Shortcut Manager should notify about conflicting shortcuts
Via a popup window and filtering by the shortcut in question to show all the conflicting ones, for example.

### PR DESCRIPTION
**Pull request by wise0704**
As @Neverbirth suggested, closes #784.

The change fixes numerous bugs caused from the previous structure, and also adds new features:
 * All conflicts are highlighted in `Color.DarkRed`.

 * View custom shortcuts without moving your mouse to click the check box. Start the filter with the key `"*"`.
image

 * View all conflicts with the filter starting keyword `"?"`.
image

 * Have options filter keywords combined in any order.
image
image

 * The dialog is now very very grumpy that it won't close until you resolve all conflicts. 
image

 * All changes are only applied when the dialog closes.

### CODE REVIEW

**Comment by wise0704** [comment]
Removing the check box and more improvements on `ShortcutDialog` comes in a [different branch](https://github.com/wise0704/flashdevelop/tree/ShortcutDialog) that will come if this PR is merged.

**Comment by wise0704** [comment]
Updates from this PR
image
image
image
image
image

Updates from the branch that can come after this:
...
...AAAAHHH I CAN'T CHECKOUT MY CURRENT BRANCH ADFJHSDFUAKHS GIT PLZ WHY

**Comment by wise0704** [comment]
After 069fda0 the dialog checks for conflicts on open.
Then I figured out that the default shortcuts have some conflicts... 
image

**Comment by Neverbirth** [comment]
I don't know about those particular cases nor how well FD allows for this but some shortcuts may be shared depending on the context and state of the application.

**Comment by wise0704** [comment]
If that is the case, then I think it only over complicates the shortcut structure...
We would need to implement a VS like structure where the dialog also lets you specify the context in a combo box, that if there is a 'Global' shortcut present it conflicts with shortcuts in other contexts...

Also I'm sure they're not shared shortcuts with different contexts. Have a look at those: (don't mind my horrible mouse-drawing skillz)
image
Pressing `Ctrl+Shift+T` only does "Undo Close".

**Comment by wise0704** [comment]
Once this PR is merged, there are some more cool things to come.

**Comment by Neverbirth** [comment]
As I said, I don't know about all of those, but for example Ctrl+F10 is that way, when you are debugging it takes precedence over Edit Macros.

**Comment by wise0704** [comment]
I think `RunToCursor` taking precedence when active is just coincidence, since the name "FlashDebugger.dll" is before "MacroManager.dll", therefore FlashDebugger registering its event handler before MacroManager. If FlashDebugger was named like NewDebugger, `RunToCursor` would never fire. 

I think if contextual shortcuts are meant to be implemented, the registration of shortcuts require a `context` parameter, which a `Global` context value takes the lowest precedence when dispatching.

**Comment by elsassph** [comment]
I think the grumpy dialog could say: There are conflicts, is that ok? <Cancel> <OK>

**Comment by Meychi** [comment]
This PR has now conflicts.

**Comment by wise0704** [comment]
@elsassph sounds good :)
but kind of a different matter but I think at least the default shortcuts shouldn't have any conflicts...

**Comment by elsassph** [comment]
True, what is in conflict atm? @Meychi ?

**Comment by wise0704** [comment]
@elsassph the word "conflict" now has conflicts...
If you meant Git conflicts, they are resolved
If you meant shorcuts, [this comment](https://github.com/fdorg/flashdevelop/pull/991#issuecomment-164221890) has the picture of the default shortcut conflicts.
Otherwise, ignore and continue :P

**Comment by wise0704** [comment]
This can also wait for #1013 to use `FindImage16()` for the button images.

File: FlashDevelop/Dialogs/ShortcutDialog.cs [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
 namespace FlashDevelop.Dialogs
 {
     public class ShortcutDialog : SmartForm
     {
-        private Timer updateTimer;
-        private ToolStripMenuItem removeShortcut;
-        private ToolStripMenuItem revertToDefault;
-        private ToolStripMenuItem revertAllToDefault;
+        const string ViewConflictsKey = "?"; // TODO: Change the type to char after #969 is merged
+        const string ViewCustomKey = "*"; // TODO: Change the type to char after #969 is merged
```
Comment (wise0704) [inline FlashDevelop/Dialogs/ShortcutDialog.cs]: Also probs wait for #969.

**Pull request merged**